server/src/ankaios_server.rs

```diff
@@ -342,6 +342,10 @@ impl AnkaiosServer {
                     self.workload_state_db
                         .process_new_states(method_obj.workload_states.clone());
 
+                    // [impl->swdd~server-removes-obsolete-delete-conditions-from-delete-graph~1]
+                    self.server_state
+                        .remove_deleted_workloads_from_delete_graph(&method_obj.workload_states);
```

Comment:
Maybe we can rename the method here to not leak out the internals of the state store.